adam/perception/perception_graph.py

```diff
@@ -60,8 +74,24 @@ def increment(self, amount=1) -> None:
 
 DebugCallableType = Callable[[DiGraph, Dict[Any, Any]], None]
 
+
+@attrs(frozen=True, eq=False, slots=True)
+class MatchedObjectNode:
```

Comment:
It was at the top of the module so that the type of `PerceptionGraphNode` could include it as a possible type. This type is defined immediately under the node.

Reply:
Got it. Can you add a module-level comment letting the reader know what are the core classes they should read first?